app/plugin.go

```diff
@@ -264,7 +270,14 @@ func (a *App) ShutDownPlugins() {
 
 	a.RemoveConfigListener(a.Srv.PluginConfigListenerId)
 	a.Srv.PluginConfigListenerId = ""
-	a.SetPluginsEnvironment(nil)
+
+	// Acquiring lock manually before cleaning up PluginsEnvironment.
+	a.Srv.PluginsLock.Lock()
+	defer a.Srv.PluginsLock.Unlock()
+	if a.Srv.PluginsEnvironment != pluginsEnvironment {
+		mlog.Warn("Another PluginsEnvironment detected while shutting down plugins.")
+	}
+	a.Srv.PluginsEnvironment = nil
```

Comment:
If we detect a different environment as such, I don't think we should `nil` it.

Reply:
Ah fair point.